### ISSUE TITLE
Seeing this error after export from eclipse: Unable to load embedded X.509 certificate

### ISSUE THREAD

**Issue by sathysreenath** [eclipse/leshan#634]
Hi,

We are trying to modify the leshan-demo-server a little bit to support our unit tests. we were able to modify it using eclipse and run it as a java application without any issue. But then when we try to get command line 'Runnable JAR file' the below error is displayed. Also the same error is displayed when no modification is done to the file and imported and exported using eclipse. We would really appreciate any help in this.

Eclipse verison:
Version: 2018-12 (4.10.0)
Build id: 20181214-0600
Java verison:
jdk-11.0.1
Windows 10

Steps for reproduction (without any change in code to the demo server):
1) import files under leshan-master into eclipse
 File -> Import-> Maven
-> Existing Maven Projects and specify the eclipse-workspace folder
2) Right click on leshan-server-demo in package explorer and select Run As->Java Application and after the server loads then terminate it.
3) Now right click on leshan-server-demo in package explorer and Export -> Java -> Runnable Jar File -> Next-> launch the java application configuration that was run and fill in the export destination, (library handling -> Extract required libraries into generated JAR)->Finish.
Here there is a warning as well:
![image](https://user-images.githubusercontent.com/36510006/50798622-9bd3ea00-129e-11e9-9689-f3c4f3394cac.png)

4) Now run the jar file saved from windows cmdline.
java -jar leshan-server-demo.jar

16:34:57.663 [main] INFO org.eclipse.californium.core.network.config.NetworkConfig - loading properties from file C:\eclipse-workspace\Californium.properties
16:34:57.670 [main] ERROR org.eclipse.leshan.server.demo.LeshanServerDemo - Unable to load embedded X.509 certificate.
java.lang.NullPointerException: null
        at org.eclipse.leshan.core.credentials.CredentialsReader.decode(CredentialsReader.java:67)
        at org.eclipse.leshan.core.credentials.CredentialsReader.readFromResource(CredentialsReader.java:47)
        at org.eclipse.leshan.server.demo.LeshanServerDemo.createAndStartServer(LeshanServerDemo.java:330)
        at org.eclipse.leshan.server.demo.LeshanServerDemo.main(LeshanServerDemo.java:234)

Also if we use the .jar file from github we do not see this issue, we are able to run the server without any issue.

Thank you.

**Comment by sbernard31** [comment]
I'm not really aware about what Eclipse do when you are using "Export -> Java -> Runnable Jar File ..."

But why do you want to use Eclipse to create your jar. Why not just using maven like explain in the [README](https://github.com/eclipse/leshan#compile-leshan--run-demos) ?

**Comment by sathysreenath** [comment]
Thanks so much, that helped. We were able to compile with maven. We will use eclipse to debug with Brakepoints and maven to compile the code. 

Only thing that we noticed was that sometimes a build failure occurs when using maven on the latest git code, this module "Integeration tests" fails. Not sure if we can skip these tests, if so that would really help in automation.

[ERROR] Errors:
[ERROR]   BootstrapTest.bootstrapToRPKServer:178 ¦ Runtime java.util.concurrent.TimeoutE...
[INFO]
[ERROR] Tests run: 133, Failures: 0, Errors: 1, Skipped: 3

Thanks once again.
